### PR TITLE
update to 1.20.2

### PR DESCRIPTION
**Pull request by adriancpp**
Hi, just cannot wait for 1.20.2 update so i made it by myself
After fixing errors i generated .jar files and tested new version on client and server on hosting - everything works!
I hope that you will also confirm that everything is updated correctly and I will be able to contribute to my favorite mod:)
Regards 
/adriancpp

### CODE REVIEW

File: src/main/java/top/seraphjack/simplelogin/network/NetworkLoader.java [hunk 1]
```diff
@@ -3,25 +3,19 @@
...
+    private static final int PROTOCOL_VERSION = 1001000;
 
-    public static SimpleChannel INSTANCE = NetworkRegistry.newSimpleChannel(
-            new ResourceLocation(SLConstants.MODID, "main"),
-            () -> PROTOCOL_VERSION,
-            PROTOCOL_VERSION::equals,
-            PROTOCOL_VERSION::equals
-    );
+    public static SimpleChannel INSTANCE = ChannelBuilder.named(
+            new ResourceLocation(SLConstants.MODID, "main")).networkProtocolVersion(PROTOCOL_VERSION).acceptedVersions(Channel.VersionTest.exact(PROTOCOL_VERSION).or(Channel.VersionTest.ACCEPT_MISSING)).clientAcceptedVersions(Channel.VersionTest.exact(PROTOCOL_VERSION).or(Channel.VersionTest.ACCEPT_MISSING)).simpleChannel();
```
Comment (SeraphJACK) [inline src/main/java/top/seraphjack/simplelogin/network/NetworkLoader.java]: Why `ACCEPT_MISSING`? I think we should forbid logging in when there's no SimpleLogin presence on the client side?

File: src/main/java/top/seraphjack/simplelogin/client/SetPasswordScreen.java [hunk 2]
```diff
@@ -55,7 +55,7 @@ protected void init() {
 
     @Override
     public void tick() {
-        this.password.tick();
+        super.tick();
```
Comment (SeraphJACK) [inline src/main/java/top/seraphjack/simplelogin/client/SetPasswordScreen.java]: Just simply remove this override :)

**Review by SeraphJACK — CHANGES_REQUESTED** [review]

**Comment by SeraphJACK** [comment]
I'll test this out tonight (hopefully)

File: src/main/java/top/seraphjack/simplelogin/network/NetworkLoader.java (hunk 1 above)
Comment (adriancpp) [reply src/main/java/top/seraphjack/simplelogin/network/NetworkLoader.java]: You're right - my mistake, i doesnt tested joining to server without mod on client side, fixing it right now

**Review by SeraphJACK — APPROVED** [review]

**Pull request merged**